.github/workflows/clippy.yml

```diff
@@ -0,0 +1,41 @@
+name: Clippy
+
+on: [push, pull_request]
+
+jobs:
+  linux-clippy:
+   name: Linux
+   runs-on: ubuntu-latest
+   steps:
+     - name: Checkout
+       uses: actions/checkout@v1
+     - name: Install Toolchain
+       uses: actions-rs/toolchain@v1
+       with:
+         components: clippy
+         override: true
+         profile: minimal
+         toolchain: 1.32.0
+     - name: Run `cargo clippy`
+       uses: actions-rs/cargo@v1
+       with:
+         command: clippy
+         args: --features "serde slog std v1 v3 v4 v5" -- -D warnings
+  windows-clippy:
+   name: Windows
+   runs-on: windows-latest
+   steps:
+     - name: Checkout
+       uses: actions/checkout@v1
+     - name: Install Toolchain
+       uses: actions-rs/toolchain@v1
+       with:
+         components: clippy
+         override: true
+         profile: minimal
+         toolchain: 1.32.0
+     - name: Run `cargo clippy`
+       uses: actions-rs/cargo@v1
+       with:
+         command: clippy
+         args: --features "guid serde slog std v1 v3 v4 v5" -- -D warnings
```

Comment:
do we need all the features activated for windows builds as well?

Reply:
Tbh it doesn’t affect the overall build time much... we can see later if we need to switch it around